pkg/apis/serving/v1beta1/revision_defaults.go

```diff
@@ -32,6 +35,28 @@ func (r *Revision) SetDefaults(ctx context.Context) {
 // SetDefaults implements apis.Defaultable
 func (rts *RevisionTemplateSpec) SetDefaults(ctx context.Context) {
 	rts.Spec.SetDefaults(ctx)
+
+	ctx = apis.WithinParent(ctx, rts.ObjectMeta)
+	if ui := apis.GetUserInfo(ctx); ui != nil {
+		ans := rts.GetAnnotations()
+		if ans == nil {
+			ans = map[string]string{}
+			defer rts.SetAnnotations(ans)
+		}
+		if apis.IsInUpdate(ctx) {
+			var oldSpec RevisionSpec
+			switch apis.GetBaseline(ctx).(type) {
+			case *Service:
+				oldSpec = apis.GetBaseline(ctx).(*Service).Spec.ConfigurationSpec.Template.Spec
+			case *Configuration:
+				oldSpec = apis.GetBaseline(ctx).(*Configuration).Spec.Template.Spec
+			}
+			if equality.Semantic.DeepEqual(oldSpec, rts.Spec) {
+				return
+			}
+		}
+		ans[serving.UpdaterAnnotation] = ui.Username
```

Comment:
Revisions are immutable, so should this be ignored in update?

Reply:
This block is for `RevisionTemplateSpec`, so I am thnking that we should update the annotation when someone updates `RevisionTemplateSpec` in ksvc (= in update).